code/go/0chain.net/smartcontract/provider/provider.go

```diff
@@ -0,0 +1,70 @@
+package provider
+
+import (
+	"fmt"
+
+	"0chain.net/core/common"
+)
+
+//go:generate msgp -io=false -tests=false -v
+
+// swagger:enum Status
+type Status int
+
+const (
+	Active Status = iota
```

Comment:
Perhaps we need to start the Status with 1 to avoid a potential issue that if some day the Status is an event filter, the default value of 0 would return all providers instead of returning `Active` providers. Please check the similar issue we had before https://github.com/0chain/0chain/issues/1381